packages/stream-node/reader.js

```diff
@@ -24,43 +26,34 @@ export const makeNodeReader = input => {
     )} to a AsyncIterator<Uint8Array>`,
   );
 
-  const finalIteration = new Promise((resolve, reject) => {
-    input.on('error', reject);
-    input.on('close', () => {
-      resolve({ done: true, value: undefined });
-    });
-  });
-
   const iterator = input[Symbol.asyncIterator]();
-  /** @type {import('@endo/stream').Reader<Uint8Array>} */
-  const reader = harden({
+  assert(iterator.return);
```

Comment:
For the purposes of getting TypeScript to play ball, the type of return is undefined or a function with a specific signature, so asserting its existence is sufficient to convince us that it’s indeed a function.

Reply:
Ugh I hate extra code for the unique purpose of satisfying the type system. If we want to be defensive, we should get `return`, assert it's a function, then do `return.call(iterator)`, in case iterator has a weird getter for `return`.

Reply:
I do not think this is a case where we need be defensive at runtime. Not yet anyway.